src/shared/varlink-io.systemd.MountFileSystem.c

```diff
@@ -60,6 +67,8 @@ static SD_VARLINK_DEFINE_METHOD(
                 SD_VARLINK_DEFINE_INPUT(password, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
                 SD_VARLINK_FIELD_COMMENT("Takes an image policy string (see systemd.image-policy(7) for details) to apply while mounting the image"),
                 SD_VARLINK_DEFINE_INPUT(imagePolicy, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
+                SD_VARLINK_FIELD_COMMENT("The mount options to use for the partitions of the image"),
+                SD_VARLINK_DEFINE_INPUT_BY_TYPE(mountOptions, PartitionMountOptions, SD_VARLINK_ARRAY|SD_VARLINK_NULLABLE),
```

Comment:
this should be SD_VARLINK_MAP keyed by the designator. Not an SD_VARLINK_ARRAY

Reply:
Could you please explain how to do the "keyed by designator" bit? There's no documentation for any of this, and no existing examples either as far as I can see, so don't know how to do it

Reply:
Also please keep in mind the options are a linked list at both ends, as that's what the service manager accepts, so it is possible to specify the options for the same 'designator' twice, it's valid input in a unit file, so it needs to work when the service manager goes through mountfsd as well, to avoid ending up in the situation where the service manager accepts inputs and can set it up by itself, but fails when going through mountfsd

Reply:
nah, allowing multiple options for the same designator is not normalized. 

IPC should only accept normalized/parsed data. Hence please pass this as map.

a map is just a json object with the keys being strings (the designator), and the value being anything (here: an object with one field for now, the option strings)